engine/input.asm

```diff
@@ -50,98 +58,46 @@ Input_VBlank:
         and #1
         bne @WaitForJoyReady
 
-    jsr Input_UpButton
-    jsr Input_DnButton
-    jsr Input_LftButton
-    jsr Input_RhtButton
+    jsr Input_Buttons
     pla
     rts
 
-Input_DnButton:
+Input_Buttons:
     pha
-    @CheckDnButton:
-        lda JOY1L                          ; check whether the Dn button was pressed this frame...
-        cmp #DNBTN
-        bne @CheckDnButtonDone 
-        A8
-        lda #1
-        sta input.inputstate.dnbtn, X
-        bra @Done
-
-    @CheckDnButtonDone:
-        A8
-        lda #0
-        sta input.inputstate.dnbtn, X
-        bra @Done
-
-    @Done:
-        A16
-        pla
-        rts
-
-Input_UpButton:
+    clc
+    lda JOY1L
+    ; Skip controller id    
+    lsr A
+    lsr A
+    lsr A
+    lsr A
+
+    ldy #12
     pha
-    @CheckUpButton:
-        lda JOY1L                          ; check whether the up button was pressed this frame...
-        cmp #UPBTN
-        bne @CheckUpButtonDone 
-        A8
-        lda #1
-        sta input.inputstate.upbtn, X
-        bra @Done
-
-    @CheckUpButtonDone:
+    txa
+    adc #input.inputstate
+    tax
+    pla
+    clc
+    @Loop:
+        lsr A
```

Comment:
I am not sure if this would work, but I think you could get a 14-cycle loop with three'ish instructions?

```
stz input.inputstate, X ; 5 cycles
lsr A                   ; 2 cycles
rol input.inputstate, X ; 7 cycles
```

Reply:
worked